wasm/jit/jit_arm64.go

```diff
@@ -554,8 +624,229 @@ func (c *arm64Compiler) compileBrTable(o *wazeroir.OperationBrTable) error {
+	// into tmpRegisters[1].
+	c.applyMemoryToRegisterInstruction(arm64.AMOVD,
+		reservedRegisterForEngine, engineGlobalContextCompiledFunctionsElement0AddressOffset,
+		tmpRegisters[1])
+
+	// Next, read the address of the target function (= &engine.compiledFunctions[offset])
+	// into compiledFunctionAddressRegister.
+	c.applyMemoryToRegisterInstruction(arm64.AMOVD,
+		tmpRegisters[1], int64(addr)*8, // * 8 because the size of *compiledFunction equals 8 bytes.
+		compiledFunctionAddressRegister)
+
+	// Finally, we are ready to write the address of the target function's *compiledFunction into the new callframe.
+	c.applyRegisterToMemoryInstruction(arm64.AMOVD,
+		compiledFunctionAddressRegister,
+		callFrameStackTopAddressRegister, callFrameCompiledFunctionOffset)
+
+	// 4) Set ra.current so that we can return back to this function properly.
+	//
+	// First, Get the return address into the tmpRegisters[2].
+	c.readInstructionAddress(obj.AJMP, tmpRegisters[2])
+	// Then write the address into the callframe.
+	c.applyRegisterToMemoryInstruction(arm64.AMOVD,
+		tmpRegisters[2],
+		// "ra.current" is BELOW the top address. See the above example for detail.
+		callFrameStackTopAddressRegister, -(callFrameDataSize - callFrameReturnAddressOffset),
+	)
+
+	// Everthing is done to make function call now.
+	// We increment the callframe stack pointer.
+	c.applyMemoryToRegisterInstruction(arm64.AMOVD,
+		reservedRegisterForEngine, engineGlobalContextCallFrameStackPointerOffset,
+		tmpRegisters[2])
+	c.applyConstToRegisterInstruction(arm64.AADD, 1, tmpRegisters[2])
+	c.applyRegisterToMemoryInstruction(arm64.AMOVD,
+		tmpRegisters[2],
+		reservedRegisterForEngine, engineGlobalContextCallFrameStackPointerOffset)
+
+	// Then, br into the target function's initial address.
+	c.applyMemoryToRegisterInstruction(arm64.AMOVD,
+		compiledFunctionAddressRegister, compiledFunctionCodeInitialAddressOffset,
+		tmpRegisters[2])
+	c.emitUnconditionalBranchToAddressOnRegister(tmpRegisters[2])
+
+	// All the registers used are temporary so we mark them unused.
+	c.markRegisterUnused(tmpRegisters...)
+
+	// We consumed the function parameters from the stack after call.
+	for i := 0; i < len(functype.Params); i++ {
+		c.locationStack.pop()
+	}
+
+	// Also, the function results were pushed by the call.
+	for _, t := range functype.Results {
+		loc := c.locationStack.pushValueLocationOnStack()
+		switch t {
+		case wasm.ValueTypeI32, wasm.ValueTypeI64:
+			loc.setRegisterType(generalPurposeRegisterTypeInt)
+		case wasm.ValueTypeF32, wasm.ValueTypeF64:
+			loc.setRegisterType(generalPurposeRegisterTypeFloat)
+		}
+	}
+
+	// On the function return, we initialize the state for this function.
+	c.initializeReservedStackBasePointerRegister()
+
+	// TODO: initialize module context, and memory pointer.
+	return nil
+}
+
+// readInstructionAddress adds an ADR instruction to set the absolute address of "target instruction"
```

Comment:
if you mean a parameter name, use it directly. You can save a sentence by clarifying it on first use

```suggestion
// readInstructionAddress adds an ADR instruction to set the absolute address of beforeTargetInst (the target instruction)
```

Reply:
no, `beforeTargetInst` is not the target and that's why I don't use the param name here.